modules/instruments/php/module.class.inc

```diff
@@ -146,5 +146,37 @@ class Module extends \Module
         }
         return [];
     }
+
+    /**
+     * Return the data dictionary for all instruments installed on a LORIS
+     * instance.
+     *
+     * @param \LORIS\LorisInstance $loris The loris instance whose dictionary
+     *                                    should be retrieved
+     *
+     * @return \LORIS\Data\Dictionary\Category[]
+     */
+    public function getDataDictionary(\LORIS\LorisInstance $loris) : iterable
+    {
+        $DB = $loris->getDatabaseConnection();
+
+        $rows = $DB->pselectCol("SELECT Test_name FROM test_names", []);
+
+        $dict = [];
+        foreach ($rows as $testname) {
+            try {
+                $inst   = \NDB_BVL_Instrument::factory($testname, "", "");
+                $cat    = new \LORIS\Data\Dictionary\Category(
+                    $testname,
+                    $inst->getFullName()
+                );
+                $fields = $inst->getDataDictionary();
+                $dict[] = $cat->withItems($fields);
+            } catch (\LorisException $e) {
+                error_log($e);
```

Comment:
are we okay with error_logging? i feel like we usually just throw the exception

Reply:
the problem with letting the exception be thrown here is that if any single instrument is broken, it breaks the entire instruments module and DQT and data dictionary module.